robottelo/products.py

```diff
@@ -316,14 +363,59 @@ class DockerRepository(BaseRepository):
     """Custom Docker repository"""
     _type = REPO_TYPE_DOCKER  # type: str
 
+    def __init__(self, url=None, distro=None, upstream_name=None):
+        self._upstream_name = upstream_name
+        super(DockerRepository, self).__init__(url=url, distro=distro)
+
+    @property
+    def upstream_name(self):
+        return self._upstream_name
+
+    def create(self, organization_id, product_id, download_policy=None, synchronize=True):
+        repo_info = make_repository({
+            'product-id': product_id,
+            'content-type': self.content_type,
+            'url': self.url,
+            'docker-upstream-name': self.upstream_name,
+        })
+        self._repo_info = repo_info
+        if synchronize:
+            self.synchronize()
+        return repo_info
+
 
 class PuppetRepository(BaseRepository):
     """Custom Puppet repository"""
     _type = REPO_TYPE_PUPPET  # type: str
 
+    def __init__(self, url=None, distro=None, modules=None):
+        # type: (str, Optional[str], List[Dict[str, str]]) -> None
+        self._puppet_modules = modules
+        super(PuppetRepository, self).__init__(url=url, distro=distro)
+
+    @property
+    def puppet_modules(self):
+        return self._puppet_modules
+
+    def add_to_content_view(self, organization_id, content_view_id):
+        options = {
+            'content-view-id': content_view_id,
+            'organization-id': organization_id,
+        }   # type: Dict[str, Any]
```

Comment:
What is this type hint for?

Reply:
no just an old implementation and forget to remove :)